instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py

```diff
@@ -261,47 +272,71 @@ def _get_view_func(request: web.Request) -> str:
         return "unknown"
 
 
-def collect_request_attributes(request: web.Request) -> dict:
-    """Collects HTTP request attributes from the ASGI scope and returns a
-    dictionary to be used as span creation attributes."""
-
-    server_host, port, http_url = (
-        request.url.host,
-        request.url.port,
-        str(request.url),
+def collect_request_attributes(
+    request: web.Request,
+    sem_conv_opt_in_mode: _StabilityMode = _StabilityMode.DEFAULT,
+) -> dict:
+    """Collects HTTP request attributes from the aiohttp request context
+    and returns a dictionary to be used as span creation attributes."""
+
+    result: dict[str, str | None] = {}
+
+    _set_http_method(
+        result,
+        request.method,
+        sanitize_method(request.method),
+        sem_conv_opt_in_mode,
     )
+    _set_http_scheme(
+        result,
+        request.scheme,
+        sem_conv_opt_in_mode,
+    )
+
+    server_host = request.url.host
+    port = request.url.port
+    if server_host:
+        _set_http_net_host(result, server_host, sem_conv_opt_in_mode)
+        if _report_old(sem_conv_opt_in_mode):
+            result[HTTP_HOST] = server_host
+    if port:
+        _set_http_net_host_port(result, port, sem_conv_opt_in_mode)
 
+    path = request.path
     query_string = request.query_string
-    if query_string and http_url:
-        if isinstance(query_string, bytes):
-            query_string = query_string.decode("utf8")
-        http_url += "?" + urllib.parse.unquote(query_string)
-
-    result = {
-        HTTP_SCHEME: request.scheme,
-        HTTP_HOST: server_host,
-        NET_HOST_PORT: port,
-        HTTP_ROUTE: _get_view_func(request),
-        HTTP_FLAVOR: f"{request.version.major}.{request.version.minor}",
-        HTTP_TARGET: request.path,
-        HTTP_URL: redact_url(http_url),
-    }
-
-    http_method = request.method
-    if http_method:
-        result[HTTP_METHOD] = http_method
-
-    http_host_value_list = (
-        [request.host] if not isinstance(request.host, list) else request.host
-    )
-    if http_host_value_list:
-        result[HTTP_SERVER_NAME] = ",".join(http_host_value_list)
-    http_user_agent = request.headers.get("user-agent")
-    if http_user_agent:
-        result[HTTP_USER_AGENT] = http_user_agent
+    target = path
+    if query_string:
+        target = f"{path}?{query_string}"
+    if target:
+        redacted_target = redact_query_parameters(target)
+        _, redacted_query = _parse_url_query(redacted_target)
+        _set_http_target(
+            result, redacted_target, path, redacted_query, sem_conv_opt_in_mode
+        )
```

Comment:
```suggestion
    if request.path_qs:
        redacted_target = redact_query_parameters(request.path_qs)
        _, redacted_query = _parse_url_query(redacted_target)
        _set_http_target(
            result, redacted_target, request.path, redacted_query, sem_conv_opt_in_mode
        )
```
Pretty sure you can use `request.path_qs` instead of manually constructing `target` with the `path` and `query_string`.

Reply:
Oh nice! Addressed in [500e41d](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3980/commits/500e41d2a6b348e0d1cce3abcde9d7a1d470ec37)